src/handle_query_contract_ui.c

```diff
@@ -6,8 +6,8 @@ static void prepend_ticker(char *dest, size_t destsize, const char *ticker) {
     if (dest == NULL || ticker == NULL) {
         THROW(0x6503);
     }
-    uint8_t ticker_len = strlen(ticker);
-    uint8_t dest_len = strlen(dest);
+    size_t ticker_len = strnlen(ticker, MAX_TICKER_LEN);
+    size_t dest_len = strnlen(dest, destsize - ticker_len);
```

Comment:
I am not sure you should use `destsize - tickerlen`. Doing this, the condition just below will always be true, and the resulting amount will be truncated.

Correct code is:
```c
    size_t dest_len = strnlen(dest, destsize);
```

Reply:
fix applied